src/ResourceManager/RecoveryServices/Commands.RecoveryServices.SiteRecovery.Test/ScenarioTests/A2A/A2ATestsHelper.ps1

```diff
@@ -0,0 +1,200 @@
+
+    
+}
+
+function getRecoveryNetworkId{
+    param([string] $location , [string] $resourceGroup)
+
+    $primaryNetworkName = "recoveryNetwork"+ $location + $seed;
+    $virtualNetwork = New-AzureRmVirtualNetwork `
+          -ResourceGroupName $resourceGroup `
+          -Location $location `
+          -Name $primaryNetworkName `
+          -AddressPrefix 10.0.0.0/16
+    $virtualNetwork.id
+}
+
+
+##
+
+<#
+.SYNOPSIS
+Wait for job completion
+Usage:
+    WaitForJobCompletion -JobId $Job.ID
+    WaitForJobCompletion -JobId $Job.ID -NumOfSecondsToWait 10
+#>
+function WaitForJobCompletion
+{ 
+    param(
+        [string] $JobId,
+        [int] $JobQueryWaitTimeInSeconds = 20,
+        [string] $Message = "NA"
+        )
+        $isJobLeftForProcessing = $true;
+        do
+        {
+            $Job = Get-AzureRmRecoveryServicesAsrJob -Name $JobId
+            Write-Host $("Job Status:") -ForegroundColor Green
+            $Job
+
+            $isJobLeftForProcessing = ($Job.State -eq 'InProgress' -or $Job.State -eq 'NotStarted')
+            
+            if($isJobLeftForProcessing)
+            {
+                if($Message -ne "NA")
+                {
+                    Write-Host $Message -ForegroundColor Yellow
+                }
+                else
+                {
+                    Write-Host $($($Job.JobType) + " in Progress...") -ForegroundColor Yellow
+                }
+                Write-Host $("Waiting for: " + $JobQueryWaitTimeInSeconds.ToString + " Seconds") -ForegroundColor Yellow
+                [Microsoft.Azure.Test.TestUtilities]::Wait($JobQueryWaitTimeInSeconds * 1000)
+            }else
+            {
+                if( !(($job.State -eq "Succeeded") -or ($job.State -eq "CompletedWithInformation")))
+                {
+                    throw "Job " + $JobId + "failed."
+                }
+            }
+        }While($isJobLeftForProcessing)
+}
+
+<#
+.SYNOPSIS
+Wait for IR job completion
+Usage:
+    WaitForJobCompletion -VM $VM
+    WaitForJobCompletion -VM $VM -NumOfSecondsToWait 10
+#>
+Function WaitForIRCompletion
+{ 
+    param(
+        [PSObject] $affectedObjectId,
+        [int] $JobQueryWaitTimeInSeconds = 10
+        )
+        $isProcessingLeft = $true
+        $IRjobs = $null
+
+        Write-Host $("IR in Progress...") -ForegroundColor Yellow
+        do
+        {
+            $IRjobs = Get-AzureRmRecoveryServicesAsrJob -TargetObjectId $affectedObjectId | Sort-Object StartTime -Descending | select -First 2 | Where-Object{$_.JobType -eq "SecondaryIrCompletion"}
+            $isProcessingLeft = ($IRjobs -eq $null -or $IRjobs.Count -ne 1)
+
+            if($isProcessingLeft)
+            {
+                Write-Host $("IR in Progress...") -ForegroundColor Yellow
+                Write-Host $("Waiting for: " + $JobQueryWaitTimeInSeconds.ToString + " Seconds") -ForegroundColor Yellow
+                [Microsoft.Azure.Test.TestUtilities]::Wait($JobQueryWaitTimeInSeconds * 1000)
```

Comment:
Should be using `[Microsoft.Rest.ClientRuntime.Azure.TestFramework.TestUtilities]::Wait`